File: src/main/java/io/prometheus/common/VersionServlet.java
package io.prometheus.common;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.PrintWriter;

public class VersionServlet
    extends HttpServlet {

  private static final long serialVersionUID = -9115048679370256251L;

  @Override
  protected void doGet(HttpServletRequest req,
                       HttpServletResponse resp)
      throws ServletException, IOException {
    resp.setStatus(HttpServletResponse.SC_OK);
    resp.setHeader("Cache-Control", "must-revalidate,no-cache,no-store");
    resp.setContentType("text/plain");
    try (final PrintWriter writer = resp.getWriter()) {
      writer.println(Utils.getVersionDesc());
    }
  }
}

File: src/main/java/io/prometheus/common/VersionServlet.kt
/*
 * Copyright © 2019 Paul Ambrose (pambrose@mac.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@file:Suppress("UndocumentedPublicClass", "UndocumentedPublicFunction")

package io.prometheus.common

import io.ktor.util.KtorExperimentalAPI
import kotlinx.coroutines.ExperimentalCoroutinesApi
import java.io.IOException
import javax.servlet.ServletException
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@KtorExperimentalAPI
@ExperimentalCoroutinesApi
class VersionServlet : HttpServlet() {

    @Throws(ServletException::class, IOException::class)
    override fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        resp.apply {
            this.status = HttpServletResponse.SC_OK
            this.setHeader("Cache-Control", "must-revalidate,no-cache,no-store")
            this.contentType = "text/plain"
            this.writer.use { it.println(getVersionDesc(true)) }
        }
    }

    companion object {
        private const val serialVersionUID = -9115048679370256251L
    }
}
